encoding.go

```diff
@@ -81,6 +81,8 @@ var (
 		6: {BitWidth: 7},
 		7: {BitWidth: 8},
 	}
+
+	extraEncodings = make(map[format.Encoding]encoding.Encoding)
```

Comment:
Should we use a `sync.Map` here or a mutex to protect access to this variable? I'm concerned that we're making `LookupEncoding` non-concurrent-safe with this change.